app/controllers/api/projects_controller.rb

```diff
@@ -105,6 +106,10 @@ def school_owner?
       school && current_user.school_owner?(school)
     end
 
+    def class_teacher?(project)
+      project.lesson_id.present? && project.lesson.school_class.present? && project.lesson.school_class.teacher_ids.include?(current_user.id)
```

Comment:
do you need `project.lesson_id.present?`? I think that's implicit by `project.lesson` existing, since the relationship wouldn't exist otherwise

Reply:
This was more to be on the safe side to cover off the cases where it's not a school project